src/index.ts

```diff
@@ -25,13 +25,16 @@ function makeBase58CheckEncoder(p2pkhVersion: number, p2shVersion: number): (dat
         ) {
           throw Error('Unrecognised address format');
         }
-        addr = Buffer.concat([Buffer.from([p2pkhVersion]), data.slice(3, 3 + data.readUInt8(2))]);
+        addr = Buffer.concat([Buffer.from(p2pkhVersion), data.slice(3, 3 + data.readUInt8(2))]);
         return bs58check.encode(addr);
       case 0xa9: // P2SH: OP_HASH160 <scriptHash> OP_EQUAL
         if (data.readUInt8(data.length - 1) !== 0x87) {
           throw Error('Unrecognised address format');
+        } else if (p2pkhVersion.length === 2) {
```

Comment:
What is the purpose of this? Why check for the length of the p2pkhVersion bytes inside the p2sh encoding branch? Can't you just use the unmodified p2shversion string in each case?